Includes response output: spo app until spo eventreceiver commands. Closes #3929 (#5540)
pnp/cli-microsoft365: docs/docs/cmd/spo/customaction/customaction-set.mdx

```md definition-list
`-u, --webUrl <webUrl>`

> : Url of the site or site collection to update the custom action..

Reviewer: ```suggestion
: Url of the site or site collection to update the custom action.
```

`-n, --name [name]`
: The name of the custom action.

`-t, --title [title]`
: The title of the custom action.

`-l, --location [location]`
: The actual location where this custom action need to be added like `CommandUI.Ribbon`.

`-g, --group [group]`
: The group of the custom action like `SiteActions`.

`-d, --description [description]`
: The description of the custom action.

`--sequence [sequence]`
: Sequence of this CustomAction being injected. Use when you have a specific sequence with which to have multiple CustomActions being added to the page.

`--actionUrl [actionUrl]`
: The URL, URI or JavaScript function associated with the action. URL example `~site/_layouts/sampleurl.aspx` or `~sitecollection/_layouts/sampleurl.aspx`.

`--imageUrl [imageUrl]`
: The URL of the image associated with the custom action.

`-e, --commandUIExtension [commandUIExtension]`
: XML fragment that determines user interface properties of the custom action.

`--registrationId [registrationId]`
: Specifies the identifier of the list or item content type that this action is associated with, or the file type or programmatic identifier.

`--registrationType [registrationType]`

> : Specifies the type of object associated with the custom action. Allowed values `None`, `List`, `ContentType`, `ProgId`, `FileType`. Default `None`.

Reviewer: ```suggestion
: Specifies the type of object associated with the custom action. Allowed values: `None`, `List`, `ContentType`, `ProgId`, `FileType`. Defaults to `None`.
```

`--rights [rights]`

> : A case-sensitive string array that contain the permissions needed for the custom action. Allowed values `EmptyMask`, `ViewListItems`, `AddListItems`, `EditListItems`, `DeleteListItems`, `ApproveItems`, `OpenItems`, `ViewVersions`, `DeleteVersions`, `CancelCheckout`, `ManagePersonalViews`, `ManageLists`, `ViewFormPages`, `AnonymousSearchAccessList`, `Open`, `ViewPages`, `AddAndCustomizePages`, `ApplyThemeAndBorder`, `ApplyStyleSheets`, `ViewUsageData`, `CreateSSCSite`, `ManageSubwebs`, `CreateGroups`, `ManagePermissions`, `BrowseDirectories`, `BrowseUserInfo`, `AddDelPrivateWebParts`, `UpdatePersonalWebParts`, `ManageWeb`, `AnonymousSearchAccessWebLists`, `UseClientIntegration`, `UseRemoteAPIs`, `ManageAlerts`, `CreateAlerts`, `EditMyUserInfo`, `EnumeratePermissions`, `FullMask`. Default `EmptyMask`.

Reviewer: ```suggestion
: A case-sensitive string array that contain the permissions needed for the custom action. Allowed values: `EmptyMask`, `ViewListItems`, `AddListItems`, `EditListItems`, `DeleteListItems`, `ApproveItems`, `OpenItems`, `ViewVersions`, `DeleteVersions`, `CancelCheckout`, `ManagePersonalViews`, `ManageLists`, `ViewFormPages`, `AnonymousSearchAccessList`, `Open`, `ViewPages`, `AddAndCustomizePages`, `ApplyThemeAndBorder`, `ApplyStyleSheets`, `ViewUsageData`, `CreateSSCSite`, `ManageSubwebs`, `CreateGroups`, `ManagePermissions`, `BrowseDirectories`, `BrowseUserInfo`, `AddDelPrivateWebParts`, `UpdatePersonalWebParts`, `ManageWeb`, `AnonymousSearchAccessWebLists`, `UseClientIntegration`, `UseRemoteAPIs`, `ManageAlerts`, `CreateAlerts`, `EditMyUserInfo`, `EnumeratePermissions`, `FullMask`. Defaults to `EmptyMask`.
```

## Examples

> Updates tenant-wide SharePoint Framework Application Customizer extension properties in the specified site.

Reviewer: ```suggestion
Updates a tenant-wide SharePoint Framework Application Customizer extension properties in the specified site.
```

```sh
m365 spo customaction set --webUrl https://contoso.sharepoint.com/sites/test --id 058140e3-0e37-44fc-a1d3-79c487d371a3 --clientSideComponentProperties '{"testMessage":"Test message"}'
```

> Updates tenant-wide SharePoint Framework **modern List View** Command Set extension in the specified site.

Reviewer: ```suggestion
Updates a tenant-wide SharePoint Framework **modern List View** Command Set extension in the specified site.
```

```sh
m365 spo customaction set --webUrl https://contoso.sharepoint.com/sites/test --id 058140e3-0e37-44fc-a1d3-79c487d371a3 --clientSideComponentProperties '{"sampleTextOne":"One item is selected in the list.", "sampleTextTwo":"This command is always visible."}' --sequence 100
```

> Updates url custom action in the SiteActions menu in the specified site.

Reviewer: ```suggestion
Updates a url custom action in the SiteActions menu in the specified site.
```

```sh
m365 spo customaction set --webUrl https://contoso.sharepoint.com/sites/test --id 058140e3-0e37-44fc-a1d3-79c487d371a3 --actionUrl "~site/SitePages/Home.aspx"
```

> Updates ScriptLink custom action with script source in **classic pages** in the specified site collection.

Reviewer: ```suggestion
Updates a ScriptLink custom action with script source in **classic pages** in the specified site collection.
```

```sh
m365 spo customaction set --webUrl https://contoso.sharepoint.com/sites/test --id 058140e3-0e37-44fc-a1d3-79c487d371a3 --scriptSrc "~sitecollection/SiteAssets/YourScript.js"
```

> Creates custom action with delegated rights in the SiteActions menu in the specified site.

Reviewer: ```suggestion
Creates a custom action with delegated rights in the SiteActions menu in the specified site.
```
